receiver/faroreceiver/payload.go

```diff
@@ -0,0 +1,384 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faroreceiver
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type Kind string
```

Comment:
Perhaps this could be part of a new [pkg/translator](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/translator)? This way, other components (in potentially other repos) could reuse this logic?